File: Chapter01/Swift/StringType.swift
//
//  StringType.swift
//  EverydayDataStructures-Swift
//
//  Created by William Smith on 8/11/16.
//  Copyright © 2016 Websmiths, LLC. All rights reserved.
//

import Foundation

class StringType
{
    static func stringDemo()
    {
        //Swift
        let one : String = "One String"
        print("One: \(one)")
        
        let two : String = "Two String"
        print("Two: \(two)")
        
        let red : String = "Red String"
        print("Red: \(red)")
        
        let blue : String = "Blue String"
        print("Blue: \(blue)")
        
        var purple : String = red + blue
        print("Concatenation: \(purple)")
        
        purple = "Purple String";
        print("Whoops! Mutation: \(purple)")
    }
}

File: Chapter01/Java/StringType.java
package main.java;

/**
 * Created by willsmith on 8/11/16.
 */
public class StringType
{
    public static void stringDemo()
    {
        //Java
        String one = "One String";
        System.out.println("One: " + one);

        String two = "Two String";
        System.out.println("Two: " + two);

        String red = "Red String";
        System.out.println("Red: " + red);

        String blue = "Blue String";
        System.out.println("Blue: " + blue);

        String purple = red + blue;
        System.out.println("Concatenation: " + purple);

        purple = "Purple String";
        System.out.println("Whoops! Mutation: " + purple);
    }
}
